plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeleteResourceHandler.java

```diff
@@ -10,46 +10,87 @@
+ * @author Jeff Maury
  */
 public class DeleteResourceHandler extends AbstractHandler {
 
 	@Override
-	public Object execute(final ExecutionEvent event) throws ExecutionException {
-		ISelection selection = HandlerUtil.getActivePart(event).getSite().getWorkbenchWindow().getSelectionService().getSelection();
-		IResource resource = UIUtils.getFirstElement(selection, IResource.class);
-		if(resource == null) {
-			IResourceUIModel model = UIUtils.getFirstElement(selection, IResourceUIModel.class);
-			if(model != null) {
-				resource = model.getResource();
-			}
-		}
-		if(resource == null) {
-			return Status.OK_STATUS;
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+	    ISelection selection = HandlerUtil.getActivePart(event).getSite().getWorkbenchWindow().getSelectionService().getSelection();
+		IResourceUIModel[] resources = UIUtils.getElements(selection, IResourceUIModel.class);
+		if(resources == null || resources.length == 0) {
+			return OpenShiftUIActivator.statusFactory().cancelStatus("No resource selected that we can delete."); //$NON-NLS-1$
 		}
-		if(MessageDialog.openQuestion(HandlerUtil.getActiveShell(event), "Delete Resource", NLS.bind("Are you certain you want to delete the {0} resource named {1}?", resource.getKind(), resource.getName()))){
-			new DeleteResourceJob(resource).schedule();
-			return Status.OK_STATUS;
-		}
-		return Status.CANCEL_STATUS;
+		String message = (resources.length > 1)?OpenShiftUIMessages.ResourceDeletionConfirmationN
+		                                       :NLS.bind(OpenShiftUIMessages.ResourceDeletionConfirmation, resources[0].getResource().getName());
+        boolean confirm = MessageDialog.openConfirm(HandlerUtil.getActiveShell(event), 
+                                                    OpenShiftUIMessages.ResourceDeletionDialogTitle, 
+                                                    message);
+        if (confirm) {
+            deleteResources(resources);
+        }
+		return null;
 	}
+	
+	/**
+	 * Resources are deleted with one job per resource and job are in a job group so
+	 * that an error dialog will be displayed at the end of the job group.
+	 * 
+	 * @param uiResources the UI resources to delete
+	 */
+	private void deleteResources(final IResourceUIModel[] uiResources) {
+	    final JobGroup group = new JobGroup("Delete Openshift resources", 1, uiResources.length) {
 
+            /*
+             * Overridden because job group cancel job at first job error by default
+             */
+            @Override
+            protected boolean shouldCancel(IStatus lastCompletedJobResult, int numberOfFailedJobs,
+                    int numberOfCanceledJobs) {
+                return false;
+            }
+	        
+	    };
+        try (Stream<IResourceUIModel> stream = Arrays.stream(uiResources)) {
+            stream.forEach(uiResource -> {
+                DeleteResourceJob job = OpenShiftJobs.createDeleteResourceJob(uiResource.getResource());
```

Comment:
This approach reuses the available job for deleting one resource. But maybe, for the sake of better user experience, it might be nice to have one job for removing several resources with the progress monitor informing user of how many resources are already removed and which resource is currently being removed.